### ISSUE TITLE
Code action items are sometimes unreadable

### ISSUE THREAD

**Issue by oliversalzburg** [microsoft/vscode#105428]
Issue Type: <b>Bug</b>

Given the following code:
```typescript
import * as GraphQL from "type-graphql";

@GraphQL.ObjectType()
export class PageInfo {
	@GraphQL.Field(is => String, {
		description: "When paginating backwards, the cursor to continue.",
		nullable: true
	})
	startCursor: string;
}
```

I receive an expected error for the `startCursor` symbol and a code action indicator.

![image](https://user-images.githubusercontent.com/1658949/91317964-fba43f00-e7ba-11ea-931c-c7d0cfb4a9db.png)

When opening the fix menu, the items are unreadable.

![image](https://user-images.githubusercontent.com/1658949/91318023-11196900-e7bb-11ea-993c-f83c257eff65.png)

VS Code version: Code 1.48.2 (a0479759d6e9ea56afa657e454193f72aef85bd0, 2020-08-25T10:13:11.295Z)
OS version: Windows_NT x64 10.0.19041
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard
Remote OS version: Linux x64 4.19.104-microsoft-standard

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.87GB (25.89GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (32.54GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (32.54GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.19.104-microsoft-standard|
|CPUs|Intel(R) Core(TM) i7-6850K CPU @ 3.60GHz (12 x 3591)|
|Memory (System)|50.11GB (32.54GB free)|
|VM|0%|
</details><details><summary>Extensions (35)</summary>

Extension|Author (truncated)|Version
---|---|---
eslint-disable-snippets|drK|1.2.0
output-colorizer|IBM|0.1.2
vscode-docker|ms-|1.5.0
remote-containers|ms-|0.134.1
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
vscode-icons|vsc|10.2.0
vscode-zipfs|arc|2.0.0
better-toml|bun|0.3.2
vscode-eslint|dba|2.1.8
xml|Dot|2.5.1
eslint-disable-snippets|drK|1.2.0
gitlens|eam|10.2.2
EditorConfig|Edi|0.15.1
prettier-vscode|esb|5.2.1
generateuuid|Fir|0.0.5
shell-format|fox|7.0.1
terraform|has|2.2.1
vscode-drawio|hed|0.7.2
vscode-esformatter|how|1.6.0
output-colorizer|IBM|0.1.2
code-beautifier|mic|2.3.3
vscode-lebab|mrm|2.6.0
vscode-puglint|mrm|2.3.0
vscode-docker|ms-|1.5.0
csharp|ms-|1.23.1
cpptools|ms-|0.29.0
hexeditor|ms-|1.2.1
js-debug-nightly|ms-|2020.8.2017
vscode-paste-image|mus|1.0.4
vscode-graphql|Pri|0.3.1
vscode-yaml|red|0.10.0
lit-plugin|run|1.2.1

(2 theme extensions excluded)

</details>

**Comment by mjbvz** [comment]
@sbatten Adding you since the screenshot looks like it us using custom context menus? Not 100% on that though.

**Comment by sbatten** [comment]
Yea if its code actions in Windows, its custom menus. It seems this menu entry has a very long label. @mjbvz is it easy for you to make a code action that would reproduce this so I can compare what native vs custom does so I can mimic it.

**Comment by mjbvz** [comment]
The code actions seems to come from an extension. I guess they may container newlines in this case?

Here's where we create them from TS: https://github.com/microsoft/vscode/blob/f9224c26cbdb13be3512a794dddc2af9fa3da2c0/extensions/typescript-language-features/src/languageFeatures/quickFix.ts#L298

Let me try adding in a few newlines to see what happens

**Comment by mjbvz** [comment]
I confirmed newlines are the problem:

<img width="484" alt="Screen Shot 2020-08-26 at 11 38 35 PM" src="https://user-images.githubusercontent.com/12821956/91400882-50fd4200-e7f5-11ea-954b-a64957d51807.png">

@sbatten Let me know if we should preprocess the code action entries or if you want to make the the custom context menus strip/replace newlines automatically

**Comment by sbatten** [comment]
The native menus on Windows do not display correctly either, it adds a special character for the new lines that doesn't look great imo. I recommend we strip them before hitting the menus

**Comment by mjbvz** [comment]
For verification:

1. On windows
1. Apply this diff to the TypeScript refactor provider:

```diff
diff --git a/extensions/typescript-language-features/src/languageFeatures/refactor.ts b/extensions/typescript-language-features/src/languageFeatures/refactor.ts
index ab0186c8f0..6f6585f12c 100644
--- a/extensions/typescript-language-features/src/languageFeatures/refactor.ts
+++ b/extensions/typescript-language-features/src/languageFeatures/refactor.ts
@@ -188,7 +188,7 @@ class InlinedCodeAction extends vscode.CodeAction {
                public readonly action: string,
                public readonly range: vscode.Range,
        ) {
-               super(title, kind);
+               super(title + '\nI hardly new ye', kind);
        }
 
        // Filled in during resolve
```

3. Now trigger any refactor in a JS/TS file

Make sure that the refactoring displays properly even though the code action has multiple lines of text

**Comment by mjbvz** [comment]
Fixed by ad1dc7f87a94edc1d06cbeb2261459f2764b2c43
